layers/src/main/scala/geotrellis/layers/hadoop/util/HdfsRangeReaderProvider.scala

```diff
@@ -0,0 +1,23 @@
+package geotrellis.layers.hadoop.util
+
+import geotrellis.layers.hadoop.HadoopCollectionLayerProvider
+import geotrellis.util.RangeReaderProvider
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+
+import java.net.URI
+
+
+class HdfsRangeReaderProvider extends RangeReaderProvider {
+  def canProcess(uri: URI): Boolean = uri.getScheme match {
+    case str: String => HadoopCollectionLayerProvider.schemes contains str.toLowerCase
+    case null => false
+  }
+
+  def rangeReader(uri: URI): HdfsRangeReader = {
+    val conf = new Configuration()
+
+    HdfsRangeReader(new Path(uri), conf)
```

Comment:
Note that you fail to call `trim` on this URI - I think this is the cause for problems with `hdfs+`

Reply:
Ah, that's a good point. I'll see if that's the cause.